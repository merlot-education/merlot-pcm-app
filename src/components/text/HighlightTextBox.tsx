import React, { ReactNode } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { ColorPallet, TextTheme } from '../../theme/theme';

export interface TextBoxProps {
  children?: ReactNode;
}

const HighlightTextBox: React.FC<TextBoxProps> = ({ children }) => {
  return (
    <View style={[style.container]}>
      <View style={[style.accentBox]} />
      <Text
        style={[
          style.headerText,
          { paddingTop: offset, paddingBottom: offset },
        ]}
        testID="HighlightTextBox"
        accessibilityLabel="HighlightTextBox"
      >
        {children}
      </Text>
    </View>
  );
};

export default HighlightTextBox;

const offset = 10;

const style = StyleSheet.create({
  icon: {
    marginRight: offset,
  },
  container: {
    flexDirection: 'row',
    backgroundColor: ColorPallet.grayscale.white,
  },
  accentBox: {
    marginRight: offset,
    backgroundColor: ColorPallet.baseColors.yellow,
    width: 8,
  },
  headerText: {
    ...TextTheme.normal,
    flexShrink: 1,
  },
});
