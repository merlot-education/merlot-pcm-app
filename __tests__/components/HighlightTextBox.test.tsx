import React from 'react'
import { create } from 'react-test-renderer'

import HighlightTextBox from '../../src/components/text/HighlightTextBox'

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')

describe('InfoTextBox Component', () => {
  it('Renders correctly', () => {
    const tree = create(
      <HighlightTextBox>Hello World</HighlightTextBox>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
