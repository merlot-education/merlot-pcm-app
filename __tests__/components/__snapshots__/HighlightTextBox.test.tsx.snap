// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTextBox Component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FCBA19",
          "marginRight": 10,
          "width": 8,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#1E9BF2",
          "flexShrink": 1,
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
